aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -544,19 +580,48 @@ private void assertNoTerminalStatus(final DBInstance dbInstance) throws CfnNotSt
 
     protected boolean isDBInstanceStabilizedAfterMutate(
             final ProxyClient<RdsClient> rdsProxyClient,
-            final ResourceModel model
+            final ResourceModel model,
+            final CallbackContext context
     ) {
         final DBInstance dbInstance = fetchDBInstance(rdsProxyClient, model);
 
         assertNoTerminalStatus(dbInstance);
 
-        return isDBInstanceAvailable(dbInstance) &&
+        if (isDBInstanceAvailable(dbInstance) &&
                 isReplicationComplete(dbInstance) &&
                 isDBParameterGroupNotApplying(dbInstance) &&
                 isNoPendingChanges(dbInstance) &&
                 isVpcSecurityGroupsActive(dbInstance) &&
                 isDomainMembershipsJoined(dbInstance) &&
-                isPromotionTierUpdated(dbInstance, model);
+                isPromotionTierUpdated(dbInstance, model)
+        ) {
+            assertNoFailedEvents(rdsProxyClient, model, context);
+            return true;
+        }
+        return false;
+    }
+
+    protected void assertNoFailedEvents(
+            final ProxyClient<RdsClient> rdsProxyClient,
+            final ResourceModel model,
+            final CallbackContext context
+    ) {
+        final Instant fetchSince = context.getTimestamp(RESOURCE_UPDATED_AT);
+        if (fetchSince == null) {
+            // we did not timestamp it before,
+            return;
+        }
+        List<Event> notifications;
+        try {
+            notifications = fetchEvents(rdsProxyClient, model, "notification", fetchSince);
+        } catch (AwsServiceException e) {
+            // best effort
```

Comment:
In the worst case we will default to current behavior. Customers will not see their stack failing, but rather would be able to discover the changes by runnng drift. I think SDK client makes a few attempts already before giving up, so the chance of it all happening during normal operation are rather slim. I think we can keep the best effort approach.

I think we should distinguish between different kinds of exceptions, though. For example, on throttling we might perform retry at a later point.